moped-editor/src/views/projects/projectsListView/ProjectsListViewTable.js

```diff
@@ -227,22 +185,63 @@ const ProjectsListViewTable = () => {
         />
         {/*Main Table Body*/}
         <Paper className={classes.paper}>
-          <Box mt={3}>
-            {!data && !error ? (
-              <CircularProgress />
-            ) : data && data.project_list_view ? (
-              <Card className={classes.root}>
-                <MaterialTable
-                  onChangeColumnHidden={handleColumnChange}
-                  data={data.project_list_view}
-                  columns={columns}
-                  title=""
-                  options={tableOptions}
-                  components={tableComponents}
-                />
-              </Card>
-            ) : (
-              <span>{error ? error : "Could not fetch data"}</span>
+          <Box
+            sx={{
+              height: "70vh",
+              minHeight: "400px",
+              padding: "14px",
+              marginTop: "24px",
+            }}
+          >
+            {data && data.project_list_view && (
+              <DataGrid
+                // per the docs: When the height of a row is set to "auto", the final height will follow exactly
+                // the content size and ignore the density. the docs recommend these styles in order to have density
+                // along with get row height auto
+                sx={{
+                  "&.MuiDataGrid-root--densityCompact .MuiDataGrid-cell": {
+                    py: "8px",
+                  },
+                  "&.MuiDataGrid-root--densityStandard .MuiDataGrid-cell": {
+                    py: "15px",
+                  },
+                  "&.MuiDataGrid-root--densityComfortable .MuiDataGrid-cell": {
+                    py: "22px",
+                  },
+                }}
+                density="compact"
+                getRowHeight={() => "auto"}
+                columnVisibilityModel={hiddenColumns}
+                onColumnVisibilityModelChange={(newModel) => {
+                  setHiddenColumns(newModel);
+                }}
+                slots={{
+                  toolbar: ProjectListToolbar,
+                }}
+                slotProps={{
+                  columnsPanel: {
+                    // Hiding buttons because when I toggle "show all" its only setting one column to visible
```

Comment:
is this a bug, or a some limitation in the way we manage column visibility?

Reply:
https://github.com/mui/mui-x/issues/11656#issuecomment-1939606662

I think its a bug